### ISSUE TITLE
Support for multi-threaded applications

### ISSUE THREAD

**Issue by saidie** [cookpad/arproxy#5]
Hi,

We rarely encountered errors like
```
Mysql2::Error: This connection is in use by: #<Celluloid::Thread:0x007f415c0262d8 sleep>
```
when running Sidekiq with Arproxy.

This seems that some Sidekiq threads simultaneously reach the following line and one thread replaces connections for others.
https://github.com/cookpad/arproxy/blob/v0.1.3/lib/arproxy/proxy_chain.rb#L34
Similar errors would be occurred for multi-threaded applications such as one using Puma.

Any support for multi-threaded applications?

**Comment by mirakui** [comment]
Sorry, currently Arproxy doesn't support multi-threaded servers like Puma.
And we don't have any milestone to implement it.
Of course, pull-requests are welcome :smiley:

**Comment by mirakui** [comment]
Released #7 as `v0.2.1`!
